test/extended/networking.sh

```diff
@@ -85,50 +99,81 @@ function save-artifacts() {
   copy-container-files "/etc/hosts" "${dest_dir}"
 }
 
-# Any non-zero exit code from any test run invoked by this script
-# should increment TEST_FAILURE so the total count of failed test runs
-# can be returned as the exit code.
-TEST_FAILURES=0
-function test-osdn-plugin() {
+function deploy-cluster() {
   local name=$1
   local plugin=$2
   local isolation=$3
+  local log_dir=$4
 
-  os::log::info "Targeting ${name} plugin: ${plugin}"
   os::log::info "Launching a docker-in-docker cluster for the ${name} plugin"
   export OPENSHIFT_NETWORK_PLUGIN="${plugin}"
   export OPENSHIFT_CONFIG_ROOT="${BASETMPDIR}/${name}"
   export OPENSHIFT_NETWORK_ISOLATION="${isolation}"
   # Images have already been built
-  export OS_DIND_BUILD_IMAGES=0
+  export OPENSHIFT_DIND_BUILD_IMAGES=0
   DIND_CLEANUP_REQUIRED=1
-  ${CLUSTER_CMD} start
-  ${CLUSTER_CMD} wait-for-cluster
 
-  os::log::info "Saving cluster configuration"
+  local exit_status=0
+
+  # Disable error checking to have a chance to save logs and artifacts
+  # if cluster deployment fails.
+  set +e
+
+  # Restart instead of start to ensure that an existing cluster is
+  # always torn down.
+  ${CLUSTER_CMD} restart
+  exit_status=$?
+
+  if [ "${exit_status}" = "0" ]; then
```

Comment:
The idiomatic way to do guarded commands when you intend to execute different code based on the output is to simply wrap the command into an `if` statement itself:
```sh
if ! ${CLUSTER_CMD} restart; then
    # failed
fi
```